qiime/core/archiver.py

```diff
@@ -96,25 +107,59 @@ def _parse_provenance(cls, provenance):
                 parameter_references=provenance['parameter-references']
             )
 
-    def __init__(self, uuid, type, provenance, archive_filepath=None):
-        self._uuid = uuid
+    def __init__(self, uuid, type, provenance, archive_filepath=None,
+                 data_initializer=None):
```

Comment:
`data_initializer` is part of a refactoring to make `Archiver` easier to reason about. It is no longer a state-machine as `save_data` is gone (and those ugly `assert` statements) in favor of this new constructor parameter.